Add auto-generated connected components in documentation (#5791)
grafana/agent: docs/README.md

If the check fails, then the doc supported services list should be updated. For that, run `make generate-cloudwatch-integration` to get the updated list, which should replace the old one in [the docs](./sources/static/configuration/integrations/cloudwatch-exporter-config.md).

## Update generated reference docs

> Some sections of Agent Flow reference documentation are automatically generated. To update them, run `make generate-docs`.

Reviewer: ```suggestion
Some sections of Grafana Agent Flow reference documentation are automatically generated. To update them, run `make generate-docs`.
```

### Community Projects

> Below is a list of community-led projects for working with Grafana Agent. These projects are not maintained or supported by Grafana Labs.

Reviewer: ```suggestion
The following is list of community-led projects for working with Grafana Agent. These projects are not maintained or supported by Grafana Labs.
```
